packages/sync/src/modules/Module.php

```diff
@@ -253,6 +253,11 @@ protected function enqueue_all_ids_as_action( $action_name, $table_name, $id_fie
 			$previous_interval_end = end( $ids );
 		}
 
+		if ( $wpdb->last_error ) {
```

Comment:
Should we be checking there is no error that is no error before we start doing our queries?

Do we need something that makes sure that we don't end up in some sort of infinite loop?

Reply:
> Should we be checking there is no error that is no error before we start doing our queries?

Performing a query clears previous errors.

> Do we need something that makes sure that we don't end up in some sort of infinite loop?

Yes, can you maybe fork this PR into a separate branch and show how this might be done?

Reply:
> Do we need something that makes sure that we don't end up in some sort of infinite loop?

@mdbitz has been playing with this fix on a very large site and hasn't seen any infinite loop, right?